hr_holidays_hour/models/hr_holidays.py

```diff
@@ -0,0 +1,256 @@
+    def onchange_date_to(self):
+        # Get result of original onchange_employee from parent class:
+        res = super(HrHolidays, self).onchange_date_to(
+            self.date_to, self.date_from
+        )
+
+        # Workaround for api incompatibility:
+        if type(res) is dict and res.has_key("value"):
+            for field, value in res.get("value").items():
+                if hasattr(self, field):
+                    setattr(self, field, value)
+
+    @api.multi
+    def _set_number_of_hours_temp(self):
+        self.ensure_one()
+        from_dt = self._compute_datetime(self.date_from)
+        to_dt = self._compute_datetime(self.date_to)
+        work_hours = self._compute_work_hours(from_dt, to_dt)
+        self.number_of_hours_temp = work_hours
+
+    @api.model
+    def _compute_datetime(self, date):
+        dt = False
+        if date:
+            this_year = datetime.date.today().year
+            reference_date = fields.Datetime.context_timestamp(
+                self.env.user, datetime.datetime(this_year, 1, 1, 12)
+            )
+            dt = fields.Datetime.from_string(date)
+            tz_dt = fields.Datetime.context_timestamp(self.env.user, dt)
+            dt = dt + tz_dt.tzinfo._utcoffset
+            dt = dt - reference_date.tzinfo._utcoffset
+        return dt
+
+    @api.multi
+    def _check_dates(self):
+        self.ensure_one()
+        # date_to has to be greater than date_from
+        if self.date_from and self.date_to:
+            if self.date_from > self.date_to:
+                raise UserError(
+                    _("The start date must be anterior to the end date.")
+                )
+
+    @api.multi
+    def _check_employee(self):
+        self.ensure_one()
+        employee = self.employee_id
+        if not employee and (self.date_to or self.date_from):
+            raise UserError(_("Set an employee first!"))
+
+    @api.multi
+    def _compute_work_hours(self, from_dt, to_dt):
+        self.ensure_one()
+        employee = self.employee_id
+        work_hours = 0.0
+        if self.date_from and self.date_to:
+            working_hours = self._get_working_hours(employee)
+            for working_hour in working_hours:
+                wh = working_hour.get_working_hours(
+                    from_dt,
+                    to_dt,
+                    compute_leaves=True,
+                    resource_id=employee.resource_id.id,
+                )
+                if wh:
+                    work_hours += wh[0]
+        return work_hours
+
+    @api.model
+    def _get_working_hours(self, employee):
+        working_hours = []
+        if employee.calendar_id:
+            working_hours.append(employee.calendar_id)
+        else:
+            contracts = employee.sudo().contract_ids
+            for contract in contracts:
+                if contract.working_hours:
+                    working_hours.append(contract.working_hours)
+        return working_hours
```

Comment:
A good pythonista is a _loop killer_. Declarative style is more readable and less bug prone. Also, you don't need to declare the variable upfront here. I'd write:

```python
if employee.calendar_id:
    working_hours = [employee.calendar_id]
else:
    working_hours = [contract.working_hours for contract in contracts if contract.working_hours]
```

Reply:
👍